pkg/account/component_test.go

```diff
@@ -51,3 +52,39 @@ func TestUpdatePassword(t *testing.T) {
 	newPasswd := "a p@55w0rd"
 	genericUpdatePasswordTest(t, oldPasswd, newPasswd, newPasswd, 1, false)
 }
+
+func TestUpdatePasswordWrongPwd(t *testing.T) {
+	oldPasswd := "wrong prev10u5"
+	newPasswd := "a p@55w0rd"
+
+	var mockCtrl = gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockKeycloakClient := mock.NewAccKeycloakClient(mockCtrl)
+	mockEventDBModule := mock.NewEventsDBModule(mockCtrl)
+	component := NewComponent(mockKeycloakClient, mockEventDBModule)
+
+	accessToken := "access token"
+	realm := "sample realm"
+	userID := "123-456-789"
+	username := "username"
+	ctx := context.WithValue(context.Background(), cs.CtContextAccessToken, accessToken)
+	ctx = context.WithValue(ctx, cs.CtContextRealm, realm)
+	ctx = context.WithValue(ctx, cs.CtContextUserID, userID)
+	ctx = context.WithValue(ctx, cs.CtContextUsername, username)
+
+	mockKeycloakClient.EXPECT().UpdatePassword(accessToken, realm, oldPasswd, newPasswd, newPasswd).Return("", fmt.Errorf("invalidPasswordExistingMessage")).Times(1)
+	mockEventDBModule.EXPECT().ReportEvent(gomock.Any(), "PASSWORD_RESET", "self-service", gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1)
+
+	err := component.UpdatePassword(ctx, oldPasswd, newPasswd, newPasswd)
+
+	assert.Equal(t, true, err != nil)
```

Comment:
assert.True(t, err!=nil)

Reply:
Done, thanks

Reply:
OMG ! Did I really suggest that ?
assert.NotNil(t, err) would have been better !